packages/components/src/slider/test/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { render, screen, fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { Slider } from '..';
+
+describe( 'Slider', () => {
+	let base;
+
+	beforeEach( () => {
+		( { container: base } = render( <Slider /> ) );
+	} );
+
+	it( 'should render correctly', () => {
+		expect( base.firstChild ).toMatchSnapshot();
+	} );
+
+	it( 'should render valuetext for unit-ed values', () => {
+		const { container } = render( <Slider defaultValue="50px" /> );
+		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
+	} );
+
+	it( 'should fire blur and focus events', async () => {
+		const onBlur = jest.fn();
+		const onFocus = jest.fn();
+		render( <Slider onBlur={ onBlur } onFocus={ onFocus } /> );
+		const slider = ( await screen.findAllByRole( 'slider' ) )[ 1 ];
+		fireEvent.focusIn( slider );
+		expect( onFocus ).toHaveBeenCalledTimes( 1 );
+		fireEvent.focusOut( slider );
+		expect( onBlur ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	it( 'should fire onChange', async () => {
+		const onChange = jest.fn();
+		render( <Slider onChange={ onChange } defaultValue="50" /> );
+		const slider = ( await screen.findAllByRole( 'slider' ) )[ 1 ];
+		fireEvent.focusIn( slider );
+		fireEvent.keyPress( slider, { key: 'ArrowUp', code: 'ArrowUp' } );
+		expect( onChange ).toHaveBeenCalledWith( '51' );
+	} );
```

Comment:
@diegohaz Any ideas why this test doesn't work? The `onChange` function doesn't get called at all 🤔

Reply:
Those `fireEvent`s will just dispatch the specific events programmatically. I don't think JSDOM will fire the change event in response to that. In fact, I'm not even sure the browser would do that either (if it's not an actual user action).

One option is firing the change event directly:
```js
fireEvent.change( slider, { target: { value: '51' } } );
```

But we can also try [`@testing-library/user-event`](https://github.com/testing-library/user-event). It looks like it's been improved a lot since the last time I proposed it. I'm not sure it supports arrow keys on range inputs though.

Reply:
Yeah I tried fiddling around with `user-event` for an hour or so and couldn't get anything workable. It might be the defaultValue issue you pointed out though

Reply:
Thanks Diego! `fireEvent.change` worked perfectly, in addition to having to pass both value and onChange.